clar2wasm/src/words/equal.rs

```diff
@@ -1254,6 +1263,17 @@ mod tests {
         crosscheck(snippet, Ok(Some(clarity::vm::Value::Bool(true))));
     }
 
+    #[test]
+    fn index_of_complex_type_versions() {
+        let snippet = "(index-of
+               (list (list (err 7))
+                     (list (ok 3)))
+               (list (err 7)))";
+        let v1 = evaluate_at(snippet, StacksEpochId::latest(), ClarityVersion::Clarity1);
+        let v2 = evaluate_at(snippet, StacksEpochId::latest(), ClarityVersion::Clarity2);
+        assert_eq!(v1, v2);
```

Comment:
Why not `crosscheck`?

Reply:
I don't think we should care about the consistency through versions. What we need is to replicate the behavior of the interpreter, so it's more important to make sure the results are identical between the interpreter and the compiled version. If the interpreter has different results through versions, we'll still need to replicate this behavior anyway.